src/core/environment/index.ts

```diff
@@ -10,7 +10,12 @@ interface ISearchParams {
 
 export default class Environment {
     private static get searchParams(): ISearchParams {
-        return new URL(window.location.href).searchParams || { get: () => null };
+        return (
+            typeof URL !== 'undefined' &&
+            URL.prototype &&
+            URL.prototype.constructor &&
+            new URL(window.location.href
+        ).searchParams) || { get: () => null };
```

Comment:
The environment / params are only used for debugging purposes and is not used by a "core" feature -- so the shim here is as lightweight as it can be. URL exists in IE 11 but it doesn't have a ctor or this feature.